composer/utils/file_helpers.py

```diff
@@ -393,13 +393,41 @@ def parse_uri(uri: str) -> tuple[str, str, str]:
         return backend, bucket_name, path.lstrip('/')
 
 
-def maybe_create_object_store_from_uri(uri: str) -> Optional[ObjectStore]:
-    """Automatically creates an :class:`composer.utils.ObjectStore` from supported URI formats.
+# Registry for object store creation functions
+object_store_registry: dict[str, Callable[[str, str], ObjectStore]] = {}
```

Comment:
Would this be nicer encapsulated in the ObjectStore class as a global variable there? 
Pros: encapsulation, Cons: it's further away from its main and only usage point `maybe_create_object_store_from_uri`